### DISCUSSION TITLE
Is there currently a way to get the previous heading of the current paragraph in markdown?

### DISCUSSION THREAD

**Discussion by luyangliuable** [remarkjs/remark#1277] [Q&A] [answered]
I am attempting to transform markdown into HTML using the marked renderer.

```js
const renderer = new marked.Renderer();
let renderedHtml: string = marked.parse(markdown, { renderer });
```

I want to adapt the paragraph rendering so that the class names of the sections match the preceding headings. Can this be achieved?

```js
renderer.paragraph = (text) => {
    return `<section>${text}</section>`;
}
```

 I am using marked and dompurify but i heard you could also to create an AST using the following syntax

```js
const file = await unified()
    .use(remarkParse)
    .use(remarkRehype)
```

My goal is to build a `table of content` section that indicates when section of the markdown is currently being observed. Without a way to correlate the paragraph to the previous heading would be hard.

**Comment by ChristianMurphy** [comment, marked as answer]
Yes, see https://www.npmjs.com/package/remark-sectionize and https://www.npmjs.com/package/@hbsnow/rehype-sectionize as an examples of how to do this.
